packages/node/src/integrations/anr/index.ts

```diff
@@ -0,0 +1,172 @@
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const inspector: InspectorApi = require('inspector');
+  let inspectorUrl: string | undefined = undefined;
+  let port = startPort;
+
+  while (inspectorUrl === undefined && port < startPort + 100) {
+    inspector.open(port);
+    inspectorUrl = inspector.url();
+    port++;
+  }
+
+  return inspectorUrl;
+}
+
+/**
+ * Starts a thread to detect App Not Responding (ANR) events
+ */
+export class Anr implements Integration {
+  public name: string = 'Anr';
+
+  public constructor(private readonly _options: Partial<Options> = {}) {}
+
+  /** @inheritdoc */
+  public setupOnce(): void {
+    //
+  }
+
+  /** @inheritdoc */
+  public setup(client: NodeClient): void {
+    if ((NODE_VERSION.major || 0) < 16) {
+      throw new Error('ANR detection requires Node 16 or later');
+    }
+
+    // setImmediate is used to ensure that all other integrations have been setup
+    setImmediate(() => this._startWorker(client));
+  }
+
+  /**
+   * Starts the ANR worker thread
+   */
+  private async _startWorker(client: NodeClient): Promise<void> {
+    const contexts = await getContexts(client);
+    const initOptions = client.getOptions();
+
+    const dsn = client.getDsn();
+
+    if (!dsn) {
+      return;
+    }
+
+    // These will not be inaccurate if sent later from the worker thread
+    delete contexts.app?.app_memory;
+    delete contexts.device?.free_memory;
+
+    const sdkMetadata = client.getSdkMetadata() || {};
+    if (sdkMetadata.sdk) {
+      sdkMetadata.sdk.integrations = client.getOptions().integrations.map(i => i.name);
+    }
+
+    const options: WorkerStartData = {
+      debug: logger.isEnabled(),
+      dsn,
+      environment: initOptions.environment || 'production',
+      release: initOptions.release,
+      dist: initOptions.dist,
+      sdkMetadata,
+      pollInterval: this._options.pollInterval || DEFAULT_INTERVAL,
+      anrThreshold: this._options.anrThreshold || DEFAULT_HANG_THRESHOLD,
+      captureStackTrace: !!this._options.captureStackTrace,
+      contexts,
+    };
+
+    if (options.captureStackTrace) {
+      startInspector();
+    }
+
+    const { Worker } = getWorkerThreads();
+
+    const worker = new Worker(new URL(`data:application/javascript;base64,${base64WorkerScript}`), {
+      workerData: options,
+    });
+    // Ensure this thread can't block app exit
+    worker.unref();
+
+    const timer = setInterval(() => {
+      try {
+        const currentSession = getCurrentHub().getScope()?.getSession();
+        // We need to copy the session object and remove the toJSON method so it can be sent to the worker
+        // serialized without making it a SerializedSession
+        const session = currentSession ? { ...currentSession, toJSON: undefined } : undefined;
+        // message the worker to tell it the main event loop is still running
+        worker.postMessage({ session });
+      } catch (_) {
+        //
+      }
```

Comment:
```suggestion
      } catch {
        // Skip errors
      }
```

Reply:
I've been bitten by this before. I don't think Typescript down levels this and some older platforms require the parameter.